platform/platform-api/src/com/intellij/ide/plugins/PluginVersionChecker.java

```diff
@@ -0,0 +1,28 @@
+// Copyright 2000-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+package com.intellij.ide.plugins;
+
+import com.intellij.openapi.extensions.ExtensionPointName;
+import com.intellij.openapi.util.Pair;
+import org.jetbrains.annotations.Nullable;
+
+public interface PluginVersionChecker {
+
+  ExtensionPointName<PluginVersionChecker> EP_NAME = new ExtensionPointName<>("com.intellij.pluginVersionChecker");
+
+  /** Returns a priority for the checker, which is used to pick a checker if there are more than one. */
+  int getPriority();
+
+  /** Returns a pair of:
+   *   the latest Kotlin plugin version for the current build number and channel;
```

Comment:
If the only purpose is Kotlin, it's a bad idea to have it in IntelliJ platform. Alternatively, this code should do something more than just an interface in the platform and it should be possible to reuse it for other plugins as well.